### DISCUSSION TITLE
OpenAI automation looping

### DISCUSSION THREAD

**Discussion by andymush** [Budibase/budibase#12513] [Help] [answered]
Hey there, 

I'd really like to ask how to implement/add the looping feature in the openAI block, as it is in the create row and other blocks. Any info or leads will be of great help. 

Thank you.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @andymush,

The reason there is no looping capabilities for this automation block is because you would get [rate limited](https://platform.openai.com/docs/guides/rate-limits/usage-tiers?context=tier-one) very quickly. Now with that being said there may be an option to allow this for self hosters but as it stands currently you won't be able to do this within Budibase's automations.

Hopefully this makes sense.
